File: src/main/kotlin/br/unipar/plano/domain/credenciamentos/usecases/clinicaHospital/impl/AtualizaPrestClinHospUseCaseImpl.java
package br.unipar.plano.domain.credenciamentos.usecases.clinicaHospital.impl;


import br.unipar.plano.domain.credenciamentos.model.clinicaHospital.PrestadorClinicaHospitalRepository;
import org.springframework.stereotype.Service;

@Service
class AtualizaPrestClinHospUseCaseImpl(private val prestadorClinicaHospitalRepository: PrestadorClinicaHospitalRepository){


}

/*
        override fun executa(idPrestadorMedico: IdPrestadorMedico, transformation: (PrestadorMedico) -> PrestadorMedico) {
        val central = prestadorMedicoRepository.findById(idPrestadorMedico).orElseThrow { PrestMedicoNotFoundException(idPrestadorMedico) }
        prestadorMedicoRepository.save(transformation(central).with(id = idPrestadorMedico))
        }

 */

File: src/main/kotlin/br/unipar/plano/domain/credenciamentos/usecases/clinicaHospital/impl/AtualizaPrestClinHospUseCaseImpl.kt
package br.unipar.plano.domain.credenciamentos.usecases.clinicaHospital.impl

import br.unipar.plano.domain.credenciamentos.model.clinicaHospital.IdPrestadorClinicaHospital
import br.unipar.plano.domain.credenciamentos.model.clinicaHospital.PrestadorClinicaHospital
import br.unipar.plano.domain.credenciamentos.model.clinicaHospital.PrestadorClinicaHospitalRepository
import br.unipar.plano.domain.credenciamentos.usecases.clinicaHospital.AtualizaPrestClinHospUseCase
import br.unipar.plano.domain.credenciamentos.usecases.prestadorMedico.impl.PrestMedicoNotFoundException
import org.springframework.stereotype.Service

@Service
class AtualizaPrestClinHospUseCaseImpl(private val prestadorClinicaHospitalRepository: PrestadorClinicaHospitalRepository): AtualizaPrestClinHospUseCase{
    override fun executa(idPrestadorClinicaHospital: IdPrestadorClinicaHospital, transformation: (PrestadorClinicaHospital) -> PrestadorClinicaHospital
    ) {
        val prestadorMedico = prestadorClinicaHospitalRepository.findById(idPrestadorClinicaHospital).orElseThrow { PrestClinHospNotFoundException(idPrestadorClinicaHospital) }
        prestadorClinicaHospitalRepository.save(transformation(prestadorMedico).with(id = idPrestadorClinicaHospital))
    }
}